test/unit/server-configmap.bats

```diff
@@ -111,3 +111,48 @@ load _helpers
       yq '.data["proxy-defaults-config.json"] | match("world") | length' | tee /dev/stderr)
   [ ! -z "${actual}" ]
 }
+
+@test "server/ConfigMap: proxyDefaults and meshGateways can be enabled" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-config-configmap.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'connectInject.centralConfig.enabled=true' \
+      --set 'connectInject.centralConfig.proxyDefaults="{\"hello\": \"world\"}"' \
+      --set 'meshGateway.enabled=true' \
+      --set 'meshGateway.globalMode=remote' \
+      --set 'client.grpc=true' \
+      . | tee /dev/stderr |
+      yq -r '.data["proxy-defaults-config.json"]' | yq -r '.config_entries.bootstrap[0].mesh_gateway.mode' | tee /dev/stderr)
+  [ "${actual}" = "remote" ]
+}
+
+@test "server/ConfigMap: proxyDefaults should have no gateway mode if disabled" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -x templates/server-config-configmap.yaml  \
+      --set 'connectInject.enabled=true' \
+      --set 'connectInject.centralConfig.enabled=true' \
+      --set 'connectInject.centralConfig.proxyDefaults="{\"hello\": \"world\"}"' \
+      --set 'meshGateway.enabled=true' \
+      --set 'meshGateway.globalMode=' \
```

Comment:
Should this be set to `null`? Or `local`?

Reply:
If set to null or empty string then it shouldn't be set. Added a test for null.